### PR TITLE
Miscellaneous editorial changes

### LINKED ISSUE #110: Imprecise wording for `POST challenge` response
`The server updates the authorization document by updating its representation of the challenge with the response fields provided by the client. The server MUST ignore any fields in the response object that are not specified as response fields for this type of challenge. The server provides a 200 response including the updated challenge.`

I think the word "including" is confusing here and might lead to belief that the spec is contradicting itself. Boulder implemented it so that the response is Authorization Resource including the updated challenge (https://github.com/letsencrypt/boulder/issues/130). I understood this as "The server provides a 200 response containing (only) the updated challenge resource", which would make it consistent with the diagram from the spec (there is an "up" link for the challenge). Since you are able to answer challenges separately, I would expect the response to be symmetrically "separate", i.e. response should be Challenge Resource instead of Authorization Resource. (cross posted from https://github.com/letsencrypt/lets-encrypt-preview/pull/362#discussion_r29407123)

### LINKED ISSUE #147: Discuss merits of acmePath in anti-replay mechanisms
In https://github.com/letsencrypt/acme-spec/pull/140, @bifurcation has objections to the `acmePath` provision that we didn't fully reach consensus on. This ticket is to discuss the alternatives.

### PR DESCRIPTION
**Pull request by bifurcation**
* Pulled out a "protocol elements" section separate from the "Cert Management" section
* Removed "Other Possibilities" under challenges section
* Light editing pass; tweaked a few SHOULDs / MUSTs
* Re-flowed some areas
* Spell check

Fixes #147 
Fixes #110

### CODE REVIEW

File: draft-barnes-acme.md [hunk 1]
```diff
@@ -1449,28 +1477,30 @@ problem document {{I-D.ietf-appsawg-http-problem}}.
...
 example:
 
-| Domain status                                 | Challenges typically sufficient for (re)Authorization |
-|:----------------------------------------------|:------------------------------------------------------|
-| No known prior certificates or ACME usage     | Domain Validation (DVSNI or Simple HTTP)              |
-| Existing valid certs, first use of ACME       | DV + Proof of Possession of previous CA-signed key    |
-| Ongoing ACME usage                            | PoP of previous Authorized key                        |
-| Ongoing ACME usage, lost Authorized key       | DV + PoP of ACME-certified Subject key                |
-| ACME usage, all keys and recovery tokens lost | Proof of legal identity of the site owner             |
+* New domain with no known certificates: Domain Validation (DVSNI or Simple
```
Comment (jcjones) [inline draft-barnes-acme.md]: Nit: Could use the Kramdown definition list construction like you did up at line 191-198.

File: draft-barnes-acme.md [hunk 2]
```diff
@@ -431,30 +370,134 @@ to, as defined in the below table:
...
+{{RFC7517}}.
+
+contact (optional, array of string):
+: An array of URIs that the server can use to contact the client for issues
+related to this authorization. For example, the server may wish to notify the
+client about server-initiated revocation.
+
+agreement (optional, string):
+: A URI referring to a subscriber agreement or terms of service provided by the
+server (see below).  Including this field indicates the client's agreement with
```
Comment (jcjones) [inline draft-barnes-acme.md]: I think this reference is dangling; I don't see (any more) a section about the ToS except here.

File: draft-barnes-acme.md (hunk 1 above)
Comment (bifurcation) [inline draft-barnes-acme.md]: Thought about that, decided not to.

File: draft-barnes-acme.md (hunk 2 above)
Comment (bifurcation) [inline draft-barnes-acme.md]: Referenced in the Registration section (grep for terms-of-service)

Comment (jcjones) [inline draft-barnes-acme.md]: Ah.

**Pull request merged**